tests/infra_test.go

```diff
@@ -200,49 +235,29 @@ var _ = Describe("Infrastructure", func() {
 			Expect(err).ToNot(HaveOccurred())
 			// we wrote data to the disk, so from now on the VM *is* running
 
-			By("Scraping the Prometheus endpoint")
-			// the VM *is* running, so we must have metrics promptly reported
-			out := getKubevirtVMMetrics(virtClient, pod, "virt-handler")
-			lines := takeMetricsWithPrefix(out, "kubevirt")
-			metrics, err := parseMetricsToMap(lines)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(metrics).To(HaveKey(ContainSubstring("kubevirt_vmi_storage_")))
-
+			metrics := collectMetrics("kubevirt_vmi_storage_")
 			By("Checking the collected metrics")
 			keys := getKeysFromMetrics(metrics)
 			for _, key := range keys {
-				if strings.HasPrefix(key, "kubevirt_vmi_storage_") && strings.Contains(key, "vdb") {
+				if strings.Contains(key, "vdb") {
 					value := metrics[key]
 					Expect(value).To(BeNumerically(">", float64(0.0)))
 				}
 			}
 		})
 
 		It("should include the network metrics for a running VM", func() {
-			By("Scraping the Prometheus endpoint")
-			// the VM *is* running, so we must have metrics promptly reported
-			out := getKubevirtVMMetrics(virtClient, pod, "virt-handler")
-			lines := takeMetricsWithPrefix(out, "kubevirt")
-			metrics, err := parseMetricsToMap(lines)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(metrics).To(HaveKey(ContainSubstring("kubevirt_vmi_network_")))
-
+			metrics := collectMetrics("kubevirt_vmi_network_")
 			By("Checking the collected metrics")
 			keys := getKeysFromMetrics(metrics)
 			for _, key := range keys {
-				if strings.HasPrefix(key, "kubevirt_vmi_network_") {
```

Comment:
I'm not sure about possible false negatives here (and yes, this should have been documented with a comment)

Reply:
Can you please explain this? the reason I removed this is because we collect all metrics whose key starts with `kubevirt_vmi_network_` (which fails if there're no such metrics reported) so in light of that, this part seems like a redundant filtering

Reply:
On second thought it is probably redundant, yes. Let's simplify things.